taca_ngi_pipeline/deliver/deliver_grus.py

```diff
@@ -452,41 +429,86 @@ def _create_delivery_project(self, pi_id, sensitive):
+            data['member_ids'] = [self.binfo_snic_id]
         response = requests.post(create_project_url, data=json.dumps(data), auth=(user, password))
         if response.status_code != 200:
             raise AssertionError("API returned status code {}. Response: {}. URL: {}".format(response.status_code, response.content, create_project_url))
         result = json.loads(response.content)
         return result
 
-    def _get_pi_id(self):
+    def _set_pi_details(self, given_pi_email=None):
+        """
+            Set PI email address and PI SNIC ID using PI email
+        """
+        self.pi_email, self.pi_snic_id = (None, None)
+        # try getting PI email
+        if given_pi_email:
+            logger.warning("PI email for project {} specified by user: {}".format(self.projectid, given_pi_email))
+            self.pi_email = given_pi_email
+        else:
+            try:
+                self.pi_email = self._get_order_detail()['project_pi_email']
+                logger.info("PI email for project {} found: {}".format(self.projectid, self.pi_email))
+            except Exception, e:
+                logger.error("Cannot fetch pi_email from StatusDB. Error says: {}".format(str(e)))
+                raise e
+        # try getting PI SNIC ID
+        try:
+            self.pi_snic_id = self._get_user_sinc_id(self.pi_email)
+            logger.info("SNIC PI-id for delivering of project {} is {}".format(self.projectid, self.pi_snic_id))
+        except Exception, e:
+            logger.error("Cannot fetch PI SNIC id using snic API. Error says: {}".format(str(e)))
+            raise e
+    
+    def _set_binfo_details(self):
+        """
+            Set bioinfo contact details if avilable, this is not mandatory so
+            the method will not raise error if it could not find any contact
+        """
+        self.binfo_email, self.binfo_snic_id = (None, None)
+        # try getting bioinfo contact email
+        try:
+            self.binfo_email = self._get_order_detail()['project_bx_email']
+            assert self.binfo_email #check its not empty string
+            logger.info("Bioinfo contact for project {} found: {}".format(self.projectid, self.binfo_email))
+        except:
+            logger.warning("Was not able to get bioinfo contact for project {}".format(self.projectid))
+        # try getting snic id for bioinfo contact
+        if self.binfo_email == self.pi_email:
+            logger.info("Bioinfo contact is same as PI, so will be ignored and not be used")
+        elif self.binfo_email:
+            try:
+                self.binfo_snic_id = self._get_user_sinc_id(self.binfo_email)
+                logger.info("SNIC id for bioinfo contact for project {} found: {}".format(self.projectid, self.binfo_snic_id))
+            except:
+                logger.warning("Was not able to get SNIC id for bioinfo contact with email {}".format(self.binfo_email))
+        
+    def _get_user_sinc_id(self, uemail):
```

Comment:
typo: sinc --> snic

Reply:
Seems like I managed to have the same typo even while calling the function 😅